lib/tennis_score.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# abstract class used to manage basic elements of scoring in
+# all stages of a tennis match
+class TennisScore
+  attr_accessor :points, :winner, :players
+
+  def initialize(min_win, win_by, type, players)
+    # create score array with scores for both teams
+    @points = [0, 0]
```

Comment:
I wonder if this would be better as a hash so it reads 

`@points = {player1: 0, player2: 0}`

currently there is no way to be sure what the two values of this array represent without going through the code.